backend/translation/views.py

```diff
@@ -182,18 +215,36 @@ def post(self, request):
                 status=status.HTTP_404_NOT_FOUND,
             )
 
-        # Return the appropriate response depending on whether a new translation was created or not
-        if created:
+        # If the translation belongs to the current user, update the translation
+        if translation.user == user:
+            translation.captions = captions
+            translation.translation_type=(
+                HUMAN_EDITED
+                if current_translation_type == MACHINE_GENERATED
+                else MANUALLY_CREATED
+            )
+            translation.save()
+            
             return Response(
-                {"message": "Translation created successfully."},
-                status=status.HTTP_201_CREATED,
+            {"message": "Translation updated successfully."}, status=status.HTTP_200_OK)
+    
+        else: 
+            new_translation = Translation.objects.create(
+                translation_type=MANUALLY_CREATED,
+                parent=translation,
+                transcript=translation.transcript,
+                target_lang=target_lang,
+                user=user,
+                payload=captions,
             )
+            
+    new_translation.save()
+    return Response(
+        {"message": "Translation created successfully."},
+        status=status.HTTP_201_CREATED,
```

Comment:
Similar to the transcript endpoint, shall we also send back the `id` in this response?

Reply:
Yeah, we should do that. I will return the `id` and `payload`.